### PR TITLE
feat(factory): add is_generator_disabled to PairConfig

### PR DESCRIPTION
**Pull request by maksk7**
add is_generator_disabled variable to PairConfig in factory

### CODE REVIEW

File: packages/astroport/src/factory.rs [hunk 1]
```diff
@@ -50,6 +50,8 @@ pub struct PairConfig {
     /// We disable pair configs instead of removing them. If it is disabled, new pairs cannot be
     /// created, but existing ones can still obtain proper settings, such as fee amounts
     pub is_disabled: Option<bool>,
+    /// Is the generator of the pair disabled
```
Comment (wedancedalot) [inline packages/astroport/src/factory.rs:53]: /// Setting to true will disable creating a pool in generator for this pair type

Comment (maksk7) [reply packages/astroport/src/factory.rs]: You're right, it's better. Changed

File: contracts/factory/src/contract.rs [hunk 2]
```diff
@@ -557,23 +557,30 @@ pub fn query_fee_info(deps: Deps, pair_type: PairType) -> StdResult<FeeInfoRespo
 ///
 /// * **_msg** is the object of type [`MigrateMsg`].
 #[cfg_attr(not(feature = "library"), entry_point)]
-pub fn migrate(deps: DepsMut, _env: Env, msg: MigrateMsg) -> Result<Response, ContractError> {
+pub fn migrate(deps: DepsMut, _env: Env, _msg: MigrateMsg) -> Result<Response, ContractError> {
     let contract_version = get_contract_version(deps.storage)?;
 
     match contract_version.contract.as_ref() {
         "astroport-factory" => match contract_version.version.as_ref() {
-            "1.0.0" => {
```
Comment (wedancedalot) [inline contracts/factory/src/contract.rs:565]: We need to think how to properly migrate this one. We have different versions deployed on testnet and mainnet, changing this part will break migration in one of them

Comment (maksk7) [reply contracts/factory/src/contract.rs:565]: It makes sense. I returned the migration for previous versions and made the MigrationMsg more universal.

File: contracts/factory/src/contract.rs [hunk 3]
```diff
@@ -575,6 +577,26 @@ pub fn migrate(deps: DepsMut, _env: Env, msg: MigrateMsg) -> Result<Response, Co
 
                 CONFIG.save(deps.storage, &new_config)?;
```
Comment (wedancedalot) [inline contracts/factory/src/contract.rs:578]: Here the contract will be migrated to the latest version as well, which is 1.2.0 but is_generator_disabled will not be added to PAIR_CONFIGS. We need to add it too

File: contracts/factory/src/contract.rs (hunk 2 above)
Comment (ShadoySV) [reply contracts/factory/src/contract.rs:565]: I agree

File: contracts/factory/src/contract.rs (hunk 3 above)
Comment (maksk7) [reply contracts/factory/src/contract.rs:578]: Done, thanks

File: contracts/factory/src/contract.rs [hunk 4]
```diff
@@ -574,6 +576,45 @@ pub fn migrate(deps: DepsMut, _env: Env, msg: MigrateMsg) -> Result<Response, Co
                 };
 
                 CONFIG.save(deps.storage, &new_config)?;
+
+                let keys = migration::PAIR_CONFIGSV110
```
Comment (wedancedalot) [inline contracts/factory/src/contract.rs:580]: it's same as on line 599, so maybe lets move to a separate migrate_pair_configs function

File: packages/astroport/src/factory.rs [hunk 5]
```diff
@@ -50,6 +50,8 @@ pub struct PairConfig {
     /// Whether a pair type is disabled or not. If it is disabled, new pairs cannot be
     /// created, but existing ones can still read the pair configuration
     pub is_disabled: Option<bool>,
+    /// Setting to true will disable creating a pool in generator for this pair type
+    pub is_generator_disabled: Option<bool>,
```
Comment (ShadoySV) [inline packages/astroport/src/factory.rs:54]: Do we need to use Option here? Maybe just use bool? Let's set both to bool by the way

File: contracts/factory/src/contract.rs (hunk 4 above)
Comment (maksk7) [reply contracts/factory/src/contract.rs]: Done, thanks.

File: packages/astroport/src/factory.rs (hunk 5 above)
Comment (maksk7) [reply packages/astroport/src/factory.rs]: It makes sense. Changed and made the default parameter `false`

File: contracts/factory/src/migration.rs [hunk 6]
```diff
@@ -45,7 +45,7 @@ pub struct PairConfigV110 {
 
 pub const PAIR_CONFIGSV110: Map<String, PairConfigV110> = Map::new("pair_configs");
 
-pub fn update_pair_configs_v110(storage: &mut dyn Storage) -> Result<(), StdError> {
+pub fn migrate_pair_configs(storage: &mut dyn Storage) -> Result<(), StdError> {
```
[the remark this answers is not in the document]
Comment (maksk7) [reply contracts/factory/src/migration.rs]: Okay, renamed.

**Review by ShadoySV — APPROVED** [review]

**Pull request merged**